packages/hardhat-vyper/src/downloader.ts

```diff
@@ -127,11 +127,15 @@ export class CompilerDownloader {
   }
 
   private _findVersionRelease(version: string): CompilerRelease | undefined {
-    return this.compilersList.find(
-      (release) =>
-        semver.valid(release.tag_name) !== null &&
-        semver.eq(release.tag_name, version)
-    );
+    return this.compilersList.find((release) => {
+      const valid = semver.valid(release.tag_name);
+      const normalized = semver.coerce(release.tag_name);
```

Comment:
don't you need to check if the coerced value is valid? `semver.valid(semver.coerce(release.tag_name));`